test_scripts/Defects/4_5/1395_Resumption_data_IGN_OFF.lua

```diff
@@ -0,0 +1,121 @@
+local runner = require('user_modules/script_runner')
+local mobile_session = require("mobile_session")
+local common = require('test_scripts/Defects/4_5/commonDefects')
+local commonFunctions = require("user_modules/shared_testcases/commonFunctions")
+local SDL = require('SDL')
+
+--[[ Local Variables ]]
+
+--[[ Local Functions ]]
+local function AddCommand(self)
+  self.mobileSession1:SendRPC("AddCommand", { cmdID = 1, vrCommands = {"VRCommand1"}})
+  EXPECT_HMICALL("VR.AddCommand"):Do(function(_, data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  self.mobileSession1:ExpectResponse("AddCommand", { success = true, resultCode = "SUCCESS" })
+  self.mobileSession1:ExpectNotification("OnHashChange")
+end
+
+local function AddCommandAfterSUSPEND(self)
+  self.hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications", { reason = "SUSPEND" })
+  self.mobileSession1:SendRPC("AddCommand", { cmdID = 2, vrCommands = {"VRCommand2"}})
+  EXPECT_HMICALL("VR.AddCommand"):Do(function(_, data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  self.mobileSession1:ExpectResponse("AddCommand", { success = true, resultCode = "SUCCESS" })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnSDLPersistenceComplete")
+  self.mobileSession1:ExpectNotification("OnHashChange"):Times(0)
+  common.delayedExp(3000)
+end
+
+local function OnAwakeSDL(self)
+  self.hmiConnection:SendNotification("BasicCommunication.OnAwakeSDL",{})
+  self.mobileSession1:ExpectNotification("OnHashChange"):Do(function(_, data)
+      self.currentHashID = data.payload.hashID
+    end)
+end
+
+local function IGNITION_OFF(self)
+  self.hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications",
+    { reason = "SUSPEND" })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnSDLPersistenceComplete"):Do(function()
+      self.hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications",
+        { reason = "IGNITION_OFF" })
+      self.mobileSession1:ExpectNotification("OnAppInterfaceUnregistered",
+        { reason = "IGNITION_OFF" })
+      EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+      EXPECT_HMINOTIFICATION("BasicCommunication.OnSDLClose")
+      SDL:DeleteFile()
+    end)
+end
+
+local function expectResumeData(self)
+  local on_vr_commands_added = EXPECT_HMICALL("VR.AddCommand"):Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS")
+      end):Times(2)
+    on_vr_commands_added:ValidIf(function(_, data)
+        if (data.params.type == "Command" and (data.params.cmdID == 1 or data.params.cmdID == 2)) then
+          if (data.params.appID == config.application1.registerAppInterfaceParams.hmi_app_id) then
+            return true
+          else
+            commonFunctions:userPrint(31, "Received the same notification or App is registered with wrong appID")
+            return false
```

Comment:
Error message can be written after `return false`, like `return false, "Received the same notification or App is registered with wrong appID"`. Minor comment.